examples/src/main/kotlin/com/teamdev/jxbrowser/examples/FileUpload.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright 2023, TeamDev. All rights reserved.
+ *
+ *  Redistribution and use in source and/or binary forms, with or without
+ *  modification, must retain the above copyright notice and the following
+ *  disclaimer.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.teamdev.jxbrowser.examples
+
+import androidx.compose.ui.window.singleWindowApplication
+import com.teamdev.jxbrowser.browser.callback.OpenFileCallback
+import com.teamdev.jxbrowser.compose.BrowserView
+import com.teamdev.jxbrowser.dsl.Engine
+import com.teamdev.jxbrowser.dsl.browser.mainFrame
+import com.teamdev.jxbrowser.dsl.register
+import com.teamdev.jxbrowser.engine.RenderingMode
+import kotlin.io.path.Path
+
+/**
+ * This example demonstrates how to register your own [OpenFileCallback]
+ * and override the functionality that displays file chooser when a user
+ * uploads a file using `INPUT TYPE="file"` HTML element on a web page.
+ */
+fun main() {
+    val engine = Engine(RenderingMode.HARDWARE_ACCELERATED)
+    val browser = engine.newBrowser()
+
+    browser.register(OpenFileCallback { _, tell ->
+        tell.open(Path("file.txt"))
+    })
+
+    browser.mainFrame?.loadHtml(
+        """
+            Please specify a file, or a set of files:<br>
+            <input type='file' name='datafile' size='40'>
+        """.trimIndent()
+    )
+
+    singleWindowApplication(title = "File Upload") {
```

Comment:
We have recently agreed to avoid title capitalisation. See the latest Material guidelines for more detail on that.